### PR TITLE
lib/core/bytes: Add a redef of has method

### PR DESCRIPTION
**Pull request by Delja**
Adding the redef of the `has` method  to take care of the negative numbers

Signed-off-by: Florian Deljarry <deljarry.florian@gmail.com>

### CODE REVIEW

File: lib/core/bytes.nit [hunk 1]
```diff
@@ -538,6 +538,12 @@ class Bytes
 		length += cln
 	end
 
+	redef fun has(c)
+	do
+		if not c isa Int then return false
```
Comment (R4PaSs) [inline lib/core/bytes.nit]: While this makes sense given the current signature, I'm wondering why we need this.

What was the reason for `has.*` functions in `Collection[E]` to not be bound to `nullable E`?

File: lib/core/bytes.nit [hunk 2]
```diff
@@ -538,6 +538,12 @@ class Bytes
...
+		return super(c&255)
```
Comment (R4PaSs) [inline lib/core/bytes.nit]: Given the semantics, maybe we should have a comment for this specialisation stating we do truncate `c` to 8 bits so we can avoid surprises in the future

File: lib/core/bytes.nit (hunk 1 above)
Comment (Delja) [reply lib/core/bytes.nit]: Bytes is an Abstract Array[Int], it’s make sense to return false when the type is not an Int for remove the list iteration. In the same case this comparison remove the type error for the int operation.

File: lib/core/bytes.nit (hunk 2 above)
Comment (privat) [reply lib/core/bytes.nit]: I agree, the behavior make sense but should be documented

File: lib/core/bytes.nit (hunk 1 above)
Comment (privat) [reply lib/core/bytes.nit]: The reason is basically to be consistent with `fun ==(nullable Object)`

note it also reduces the number of dynamic type errors: better return false than aborting.

~~~nit
var a: Array[Object] = [1,2,3]
print a.has("toto")
# there is obviously no toto, I will tell you, I won't crash your program for that
~~~

**Pull request merged**